sdks/python/apache_beam/runners/direct/transform_evaluator.py

```diff
@@ -65,16 +71,23 @@
 from apache_beam.utils.timestamp import MIN_TIMESTAMP
 from apache_beam.utils.timestamp import Timestamp
 
+if TYPE_CHECKING:
+  from apache_beam.io.gcp.pubsub import _PubSubSource
```

Comment:
These kinds of imports mean one would have to install apache_beam[gcp] to to type checking, right? Alternatively we could declare its superclass, or omit it.

Reply:
Good point.  
mypy will only inspect dependencies that declare that they are typed via [a py.typed file](https://www.python.org/dev/peps/pep-0561/), so assuming some of those gcp deps are pep-561-compliant, then, yes, the mypy type analysis will be more accurate with them installed.  I can add them to the tox env.

Reply:
I forgot to mention, it is not an error for those types to be missing, unless you enable it (which I did not do).  If stubs or typed source for that class is not available it effectively becomes `Any`